### PR TITLE
fix: CREATE ROLE role1 error

### LINKED ISSUE #12235: bug:  CREATE ROLE role1 error
**Summary**

```
CREATE ROLE role1;
```

Error:
```
1005:
error: 
  --> SQL:3:13
  |
1 | 
2 | 
3 | CREATE ROLE role1;
  | ------      ^^^^^ expected  or `IF`
  | |            
  | while parsing `CREATE ROLE [IF NOT EXISTS] '']`
```

Doc: https://databend.rs/doc/sql-commands/ddl/user/user-create-role

### PR DESCRIPTION
**Pull request by parkma99**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
 replace the String token into an Identifier token in `create_role`

- Closes #12235

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12341)

### CODE REVIEW

**Comment by BohuTANG** [comment]
Great, but we need a logic test for this changed.

**Comment by parkma99** [comment]
Thanks, Now I am at home this week, I will go to office next week.

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -879,7 +879,7 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
     let show_roles = value(Statement::ShowRoles, rule! { SHOW ~ ROLES });
     let create_role = map(
         rule! {
-            CREATE ~ ROLE ~ ( IF ~ NOT ~ EXISTS )? ~ #literal_string
+            CREATE ~ ROLE ~ ( IF ~ NOT ~ EXISTS )? ~ #parameter_to_string
```
Comment (BohuTANG) [inline src/query/ast/src/parser/statement.rs:882]: all `ROLE` related opeartions should be `parameter_to_string`, such as `DROP ROLE`...

Comment (TCeason) [reply src/query/ast/src/parser/statement.rs:882]: image

Yes. drop/create role, grant/revoke role, set role,

Comment (TCeason) [reply src/query/ast/src/parser/statement.rs:882]: the easy way, you need to all `ROLE ~ #literal_string`.

File: src/query/ast/src/parser/statement.rs [hunk 2]
```diff
@@ -1504,8 +1504,8 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
             | #alter_user : "`ALTER USER ('<username>'@'hostname' | USER()) [IDENTIFIED [WITH <auth_type>] [BY <password>]] [WITH <user_option>, ...]`"
             | #drop_user : "`DROP USER [IF EXISTS] '<username>'@'hostname'`"
             | #show_roles : "`SHOW ROLES`"
-            | #create_role : "`CREATE ROLE [IF NOT EXISTS] '<role_name>']`"
-            | #drop_role : "`DROP ROLE [IF EXISTS] '<role_name>'`"
+            | #create_role : "`CREATE ROLE [IF NOT EXISTS] <role_name>]`"
```
Comment (xudong963) [inline src/query/ast/src/parser/statement.rs:1507]:
```suggestion
            | #create_role : "`CREATE ROLE [IF NOT EXISTS] <role_name>`"
```

File: src/query/ast/tests/it/parser.rs [hunk 3]
```diff
@@ -175,8 +175,8 @@ fn test_statement() {
         r#"create user 'test-e' identified by 'password';"#,
         r#"drop user if exists 'test-j';"#,
         r#"alter user 'test-e' identified by 'new-password';"#,
-        r#"create role 'test'"#,
```
[the remark this answers is not in the document]
Comment (BohuTANG) [reply src/query/ast/tests/it/parser.rs:178]: Not expected, this must work.

[the remark this answers is not in the document]
Comment (TCeason) [reply src/query/ast/tests/it/parser.rs:178]: Already compatible role_name with lit string.

**Pull request merged**